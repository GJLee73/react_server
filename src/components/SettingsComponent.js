import React from 'react';

const SettingsComponent = ({info,push,quit}) => {
    
    
    console.log(info)
    const userInfo = info.toJS()
    console.log(userInfo,'hi')
    return (<div>
        <div className="container auth">
				
				<div className="card">
					<div className="header blue white-text center">
						<div className="card-content">개인정보</div>
					</div>
				
                <div className="card-content">
                <div className="row">
                <div style={{cursor:'pointer'}} onClick={()=>{push('nickname')} }>
                    <div className="input-field col s6">
                    닉네임:
                    </div>
                    <div className="input-field col s6">
                        <div >
                            {userInfo.nickname}
                        </div>
                    </div>
                </div>
                <div style={{cursor:'pointer'}} onClick={()=>{push('password')}}>
                    <div className="input-field col s12">
                    비밀번호 수정
                    </div>
                </div>
                <div style={{cursor:'pointer'}} onClick={()=>{push('email')}}>
                    <div className="input-field col s6">
                    이메일:
                    </div>
                    <div className="input-field col s6">
                        <div >
                            {userInfo.email}
                        </div>
                    </div>
                </div>
                
               <div style={{cursor:'pointer'}} onClick={()=>{push('phone')}}>
                    <div className="input-field col s6">
                    휴대폰:
                    </div>
                    <div className="input-field col s6">
                        <div >
                            {userInfo.phone}
                        </div>
                    </div>
                </div>
                
                         </div>
        
                    </div>

                    <button onClick={quit}>탈퇴하기</button>
                </div>
            </div>
    </div>)
}
 
export default SettingsComponent;