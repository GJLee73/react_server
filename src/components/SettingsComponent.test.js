import React from 'react';
import {shallow} from 'enzyme'
import SettingsComponent from './SettingsComponent'
import {Map} from 'immutable'

describe('settingsComponent test',()=>{
    var push = jest.fn()
    var quit = jest.fn()
    var info = Map({nickname:'g1',email:'jwc2094@naver.com',phone:'000-0000-0000'})
    var wrapper =shallow(<SettingsComponent push={push} quit={quit} info={info}/>)
    var divs = wrapper.find('.row> div')
    it('click test',()=>{
        divs.at(0).simulate('click')
        divs.at(1).simulate('click')
        divs.at(2).simulate('click')
        divs.at(3).simulate('click')
        expect(push.mock.calls.length).toBe(4)
    })
    it('snapShot',()=>{
        expect(wrapper).toMatchSnapshot()
    })
})