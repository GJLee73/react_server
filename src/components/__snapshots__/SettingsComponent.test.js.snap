// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settingsComponent test snapShot 1`] = `
<div>
  <div
    className="container auth"
  >
    <div
      className="card"
    >
      <div
        className="header blue white-text center"
      >
        <div
          className="card-content"
        >
          개인정보
        </div>
      </div>
      <div
        className="card-content"
      >
        <div
          className="row"
        >
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="input-field col s6"
            >
              닉네임:
            </div>
            <div
              className="input-field col s6"
            >
              <div>
                g1
              </div>
            </div>
          </div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="input-field col s12"
            >
              비밀번호 수정
            </div>
          </div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="input-field col s6"
            >
              이메일:
            </div>
            <div
              className="input-field col s6"
            >
              <div>
                jwc2094@naver.com
              </div>
            </div>
          </div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="input-field col s6"
            >
              휴대폰:
            </div>
            <div
              className="input-field col s6"
            >
              <div>
                000-0000-0000
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        onClick={[Function]}
      >
        탈퇴하기
      </button>
    </div>
  </div>
</div>
`;
